### PR TITLE
Styling lint rule for CSS attributes kebab case

### PR DESCRIPTION
**Pull request by libertymayc**
Part of CSS clean up and standardisation:

- Standardise attributes appearing in tokens to always be kebab case, e.g. -border-color -> -borderColor (lint rule added)

### CODE REVIEW

File: packages/core/src/checkbox/CheckboxIcon.css [hunk 1]
```diff
@@ -24,10 +24,10 @@
   --checkbox-borderWidth: var(--uitk-selectable-borderWidth);
   --checkbox-tick-fill: var(--uitkCheckboxIcon-tick-color, var(--uitk-selectable-primary-foreground-selected));
   --checkbox-viewbox: 14px; /* do not change, matches value in svg viewBox */
-  --checkbox-box-size: calc(var(--checkbox-viewbox) - var(--checkbox-borderWidth));
+  --checkbox-size: calc(var(--checkbox-viewbox) - var(--checkbox-borderWidth));
   --checkbox-box-offset: calc(var(--checkbox-borderWidth) / 2);
 
-  fill: var(--uitkCheckboxIcon-solid-color, var(--uitk-selectable-background));
```
Comment (joshwooding) [inline packages/core/src/checkbox/CheckboxIcon.css:30]: Some of these changes don't seem to be only about kebab case?

Comment (libertymayc) [reply packages/core/src/checkbox/CheckboxIcon.css:30]: Moved to #346

File: packages/core/src/checkbox/CheckboxIcon.css [hunk 2]
```diff
@@ -24,7 +24,7 @@
   --checkbox-borderWidth: var(--uitk-selectable-borderWidth);
   --checkbox-tick-fill: var(--uitkCheckboxIcon-tick-color, var(--uitk-selectable-primary-foreground-selected));
   --checkbox-viewbox: 14px; /* do not change, matches value in svg viewBox */
```
[the remark this answers is not in the document]
Comment (libertymayc) [reply packages/core/src/checkbox/CheckboxIcon.css:26]: I'm not sure - i think this came from @heswell if steve has reasoning?

File: packages/theme/css/characteristics/editable.css [hunk 3]
```diff
@@ -22,7 +22,7 @@
   --uitk-editable-borderColor-active: var(--uitk-palette-interact-border-active);
   --uitk-editable-borderColor-disabled: var(--uitk-palette-interact-border-disabled);
   --uitk-editable-borderColor-hover: var(--uitk-palette-interact-border-hover);
-  --uitk-editable-borderColor-readonly: var(--uitk-palette-interact-border-readonly);
+  --uitk-editable-borderColor-readonly: var(--uitk-palette-interact-border-readOnly);
```
Comment (alycrys) [inline packages/theme/css/characteristics/editable.css:25]: `readonly` attribute is all lowercase

File: packages/theme/css/ToolkitIcons.css [hunk 4]
```diff
@@ -2,7 +2,7 @@
...
...oADQADAAEECQACAA4CvAADAAEECQADABoCdAADAAEECQAEABoC1wADAAEECQAFABYCUQADAAEECQAGABoCmwADAAEECQAJADoBLgADAAEECQAKAGYAqwADAAEECQALADYAQgADAAEECQAMADQBgmpwbXVpdGstaWNvbnMAagBwAG0AdQBpAHQAawAtAGkAYwBvAG4Ac2h0dHA6Ly9nby5qcG1jaGFzZS5uZXQvdWl0awBoAHQAdABwADoALwAvAGcAbwAuAGoAcABtAGMAaABhAHMAZQAuAG4AZQB0AC8AdQBpAHQAa0pQTSBVSSBUb29sa2l0IEljb24gRm9udApGb250IGdlbmVyYXRlZCBieSBJY29Nb29uLgBKAFAATQAgAFUASQAgAFQAbwBvAGwAawBpAHQAIABJAGMAbwBuACAARgBvAG4AdAAKAEYAbwBuAHQAIABnAGUAbgBlAHIAYQB0AGUAZAAgAGIAeQAgAEkAYwBvAE0AbwBvAG4ALkNJQiBEaWdpdGFsIEV4cGVyaWVuY2UgRGVzaWduAEMASQBCACAARABpAGcAaQB0AGEAbAAgAEUAeABwAGUAcgBpAGUAbgBjAGUAIABEAGUAcwBpAGcAbmh0dHA6Ly9nby5qcG1jaGFzZS5uZXQvZHhkAGgAdAB0AHAAOgAvAC8AZwBvAC4AagBwAG0AYwBoAGEAcwBlAC4AbgBlAHQALwBkAHgAZKkgMjAyMCBKUE1vcmdhbiBDaGFzZSAmIENvLiBBbGwgcmlnaHRzIHJlc2VydmVkLgCpACAAMgAwADIAMAAgAEoAUABNAG8AcgBnAGEAbgAgAEMAaABhAHMAZQAgACYAIABDAG8ALgAgAEEAbABsACAAcgBpAGcAaAB0AHMAIAByAGUAcwBlAHIAdgBlAGQALlZlcnNpb24gMy4xAFYAZQByAHMAaQBvAG4AIAAzAC4AMWpwbXVpdGstaWNvbnMAagBwAG0AdQBpAHQAawAtAGkAYwBvAG4Ac2pwbXVpdGstaWNvbnMAagBwAG0AdQBpAHQAawAtAGkAYwBvAG4Ac1JlZ3VsYXIAUgBlAGcAdQBsAGEAcmpwbXVpdGstaWNvbnMAagBwAG0AdQBpAHQAawAtAGkAYwBvAG4AcwAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==")
```
Comment (alycrys) [inline packages/theme/css/ToolkitIcons.css:5]: Not sure this is correct

File: packages/theme/css/foundations/fade.css [hunk 5]
```diff
@@ -18,7 +18,7 @@
   --uitk-color-grey-300-fade-border: rgba(76, 80, 91, var(--uitk-palette-opacity-border));
   --uitk-color-grey-600-fade-border: rgba(47, 49, 54, var(--uitk-palette-opacity-border));
 
-  --uitk-color-grey-90-fade-border-readonly: rgba(132, 135, 142, var(--uitk-palette-opacity-border-readOnly));
+  --uitk-color-grey-90-fade-border-readOnly: rgba(132, 135, 142, var(--uitk-palette-opacity-border-readOnly));
```
Comment (alycrys) [inline packages/theme/css/foundations/fade.css:21]: I wouldn't change this one

File: packages/theme/css/characteristics/overlayable.css [hunk 6]
```diff
@@ -1,7 +1,7 @@
 .uitk-light,
 .uitk-dark {
   --uitk-overlayable-shadow-scroll: var(--uitk-shadow-1);
-  --uitk-overlayable-shadow-border-region: var(--uitk-shadow-2);
+  --uitk-overlayable-shadow-borderRegion: var(--uitk-shadow-2);
```
Comment (alycrys) [inline packages/theme/css/characteristics/overlayable.css:4]: hmmm, is this valid?

Comment (libertymayc) [reply packages/theme/css/characteristics/overlayable.css:4]: this is how it is in Figma

Comment (alycrys) [reply packages/theme/css/characteristics/overlayable.css:4]: typo?

Comment (libertymayc) [reply packages/theme/css/characteristics/overlayable.css:4]: this came from design, we together decided to make any subitems/concepts camel case for easier reading and separation of token part semantics (see e.g. size foundation for examples)

File: .changeset/itchy-peaches-unite.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+"@jpmorganchase/uitk-core": minor
+"@jpmorganchase/uitk-grid": minor
+"@jpmorganchase/uitk-icons": minor
+"@jpmorganchase/uitk-lab": minor
+"@jpmorganchase/uitk-theme": minor
+---
+
+Use lint rule to ensure CSS attributes in token are kebab case, e.g. -border-color to -borderColor
```
[the remark this answers is not in the document]
Comment (libertymayc) [reply .changeset/itchy-peaches-unite.md:9]: planning to update this!

File: packages/theme/stories/foundations/palette.stories.mdx [hunk 8]
```diff
@@ -86,7 +86,7 @@ Opacity values to adjust different attributes.
       />
       <OpacityBlock
         opacity="--uitk-opacity-1"
-        cssVariable="--uitk-palette-opacity-border-readOnly"
+        cssVariable="--uitk-palette-opacity-border-readonly"
```
[the remark this answers is not in the document]
Comment (libertymayc) [reply packages/theme/stories/foundations/palette.stories.mdx:89]: i changed re @alycrys first comment on her review - but thats a good point too.. i think its different in a lot of places

**Review by origami-z — APPROVED** [review]

**Review by alycrys — APPROVED** [review]

**Pull request merged**